library/VTKExtensions/Rendering/vtkF3DRenderer.cxx

```diff
@@ -498,131 +502,152 @@ void vtkF3DRenderer::SetHDRIFile(const std::string& hdriFile)
-      if (!vtksys::SystemTools::FileExists(this->HDRIFile, true))
-      {
-        F3DLog::Print(
-          F3DLog::Severity::Warning, std::string("HDRI file does not exist ") + this->HDRIFile);
-      }
-      else
+      F3DLog::Print(
+        F3DLog::Severity::Warning, std::string("HDRI file does not exist ") + this->HDRIFile);
+    }
+    else
+    {
+      auto reader = vtkSmartPointer<vtkImageReader2>::Take(
+        vtkImageReader2Factory::CreateImageReader2(this->HDRIFile.c_str()));
+      if (reader)
       {
-        auto reader = vtkSmartPointer<vtkImageReader2>::Take(
-          vtkImageReader2Factory::CreateImageReader2(this->HDRIFile.c_str()));
-        if (reader)
-        {
-          reader->SetFileName(this->HDRIFile.c_str());
-          reader->Update();
-
-          hdriTexture->SetColorModeToDirectScalars();
-          hdriTexture->MipmapOn();
-          hdriTexture->InterpolateOn();
-          hdriTexture->SetInputConnection(reader->GetOutputPort());
+        reader->SetFileName(this->HDRIFile.c_str());
+        reader->Update();
 
-          // 8-bit textures are usually gamma-corrected
-          if (reader->GetOutput() && reader->GetOutput()->GetScalarType() == VTK_UNSIGNED_CHAR)
-          {
-            hdriTexture->UseSRGBColorSpaceOn();
-          }
+        hdriTexture->SetColorModeToDirectScalars();
+        hdriTexture->MipmapOn();
+        hdriTexture->InterpolateOn();
+        hdriTexture->SetInputConnection(reader->GetOutputPort());
 
-          this->HasHDRI = true;
-        }
-        else
+        // 8-bit textures are usually gamma-corrected
+        if (reader->GetOutput() && reader->GetOutput()->GetScalarType() == VTK_UNSIGNED_CHAR)
         {
-          F3DLog::Print(
-            F3DLog::Severity::Warning, std::string("Cannot open HDRI file ") + this->HDRIFile);
+          hdriTexture->UseSRGBColorSpaceOn();
         }
+
+        this->HasHDRILighting = true;
+      }
+      else
+      {
+        F3DLog::Print(
+          F3DLog::Severity::Warning, std::string("Cannot open HDRI file ") + this->HDRIFile);
       }
     }
+  }
 
-    // Dynamic HDRI
-    if (this->HasHDRI)
-    {
+  // Dynamic HDRI
+  if (this->HasHDRILighting)
+  {
 #if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 2, 20221220)
-      // Check LUT cache
-      std::string lutCachePath = this->CachePath + "/lut.vti";
-      bool lutCacheExists = vtksys::SystemTools::FileExists(lutCachePath, true);
-      if (lutCacheExists)
+    // Check LUT cache
+    std::string lutCachePath = this->CachePath + "/lut.vti";
+    bool lutCacheExists = vtksys::SystemTools::FileExists(lutCachePath, true);
+    if (lutCacheExists)
+    {
+      // TODO: this->EnvMapLookupTable should be a smart pointer in VTK
+      if (this->EnvMapLookupTable)
       {
-        vtkF3DCachedLUTTexture* lut = vtkF3DCachedLUTTexture::New();
-        lut->SetFileName(lutCachePath.c_str());
-        this->EnvMapLookupTable = lut;
+        this->EnvMapLookupTable->Delete();
```

Comment:
We had a leak here before?

Reply:
yes, when switching file using cached HDRIs

Reply:
ok

Reply:
why hasn't it been caught by the sanitizer?

Reply:
TBH I'm not sure, HDRIChange should have caught it. I wont be able to investigate right now.

Reply:
Actually HDRIChange did not catch it because the renderer is created from scratch when swithing drom default scene to full scene.

Reply:
New test would catch it.